forwarding.go

```diff
@@ -40,7 +40,12 @@ func (r *oauthProxy) proxyMiddleware(next http.Handler) http.Handler {
 		}
 
 		// @step: add the proxy forwarding headers
-		req.Header.Add("X-Forwarded-For", realIP(req))
+		req.Header.Set("X-Real-IP", realIP(req))
+		if xff := req.Header.Get(headerXForwardedFor); xff == "" {
+			req.Header.Set("X-Forwarded-For", realIP(req))
```

Comment:
This doesn't seem right.  It should append, not replace.

Reply:
It does:
1) If XFF exists in request, forward it to app behind proxy as is.
2) If no XFF received in request, set it equal to client's real IP.

Existing implementation adds new header, so app behind proxy receives two different XFF headers - original one and new with just client's real ip, which is definitely not right too :)

Maybe i missed different cases.. will check.